lib/compiler/passes/generate-bytecode.js

```diff
@@ -197,18 +196,42 @@ const visitor = require("../visitor");
 //
 //        silentFails--;
 function generateBytecode(ast) {
-  const consts = [];
+  const literals = [];
+  const classes = [];
+  const expectations = [];
+  const functions = [];
 
-  function addConst(value) {
-    const index = consts.indexOf(value);
+  function addLiteralConst(value) {
+    const index = literals.indexOf(value);
 
-    return index === -1 ? consts.push(value) - 1 : index;
+    return index === -1 ? literals.push(value) - 1 : index;
   }
 
-  function addFunctionConst(params, code) {
-    return addConst(
-      "function(" + params.join(", ") + ") {" + code + "}"
-    );
+  function addClassConst(node) {
+    const cls = {
+      value: node.parts,
+      inverted: node.inverted,
+      ignoreCase: node.ignoreCase
+    };
+    const pattern = JSON.stringify(cls);
+    const index = classes.findIndex(c => JSON.stringify(c) === pattern);
+
+    return index === -1 ? classes.push(cls) - 1 : index;
+  }
+
+  function addExpectedConst(expected) {
+    const pattern = JSON.stringify(expected);
+    const index = expectations.findIndex(e => JSON.stringify(e) === pattern);
+
```

Comment:
Same as classes above.  what about expections as object mapping JSON to expected?

Reply:
The same answer